openpype/hosts/houdini/api/lib.py

```diff
@@ -836,6 +836,70 @@ def get_current_context_template_data_with_asset_data():
     return template_data
 
 
+def get_houdini_color_settings():
+    """Get Houdini working file color settings.
+
+    Returns:
+        Dict: The dictionary contains the Houdini working file color settings
+              if the settings are enabled, otherwise it is an empty dictionary.
+    """
+
+    project_settings = get_current_project_settings()
+    color_settings = project_settings["houdini"]["imageio"]["workfile"]
+    if color_settings["enabled"]:
+        color_settings.pop("enabled")
+        # Remove leading, and trailing whitespaces
+        view_Space = color_settings["review_color_space"]
+        color_settings["review_color_space"] = view_Space.strip()
+        return color_settings
+
+    return {}
+
+
+def set_review_color_space(opengl_node, log=None):
```

Comment:
isn't setting `OCIO_ACTIVE_DISPLAYS` and `OCIO_ACTIVE_VIEWS` enough for this to work? this worries me that we are perhaps overriding the Houdini native OCIO workflow which relies solely on setting OCIO and the other OCIO env vars for it to work

Reply:
I think those are two different thing, _please correct me_: 
- Setting default display and view via setting `OCIO_ACTIVE_DISPLAYS` and `OCIO_ACTIVE_VIEWS`
- Picking output transform for review 

before this PR, 
`set_review_color_space` function only used the default output transform that's typically equivalent to scene view. 

but if someone for some reason wanted to specify an output transform other than the default ? what if someone wants the value to be explicitly shown ?

This's why I added review transform setting in this PR. 

--- 
 
This test https://github.com/ynput/OpenPype/pull/5322#issuecomment-1677259981 shows why we should pick an output transform for opengl nodes.

Reply:
`OCIO` env var is already being set by global pre launch hook. This is an extension to that.

Reply:
yeah I see, a few DCCs make use of the OCIO native env vars like `OCIO_ACTIVE_DISPLAYS` and `OCIO_ACTIVE_VIEWS` to control common OCIO workflows and fill up the default values, however I do see that Houdini doesn't make use of that on the `OpenGL` node yet (even though it might add support for it in the future) so this PR makes sense to control that.

Reply:
we are currently setting `OCIO_ACTIVE_DISPLAYS` and `OCIO_ACTIVE_VIEWS` in the env of our Applications directly to control the defaults on the viewers (for both Nuke and Houdini), that's why I'm saying this that I wouldn't want OP to hijack that existing vanilla OCIO workflow

Reply:
I think the system is flexible as 
1. you can set `OCIO_ACTIVE_DISPLAYS` and `OCIO_ACTIVE_VIEWS` using env vars in Houdini's and Nuke app settings 
2. you can depend on Nuke color management settings (that update the settings inside nuke file)
3. you can depend on Houdini color management settings (by settings env vars for the launched houdini session)

I can tell that no.1 overrides no.2, correct ? 
Do you think that no.1 and no.2 can conflict ? if yes, do you have suggestions how to solve that conflict ?

Reply:
The issue fabia is referring to that 1 and 3 are in conflict.

Because 3) will set the env vars on launch, thus overriding any env vars that might have been set elsewhere - e.g. in application environments.

To avoid that - there might need to be an explicit enable checkbox to "Set vars on launch" with maybe a tooltip that explains that when disabled it's up to the admin to manage the vars manually. That way you can disable setting the vars on launch via the pre launc hook, allow custom overrides via Application settings and still allow to toggle the "Validate Review Colorspace"

**However** in this instance I think @fabiaserra would actually be better off removing his vars from the Application settings and instead put it in the project settings like here so that each project can have its own defined settings for it. Unless Fabia has e.g. defined multiple applications defined like "houdini_srgb" and "houdini_linear" or whatever so that the user can actually run different OCIO configs via that way but that sounds like an anti-pattern as well (even though that's technically not possible with the project settings implementation; you couldn't have multiple definitions within a project (also not _per asset!_)

Reply:
yeah, we have 
- global env vars
- application env vars
- application version env vars
- tools (per project per asset)
- houdini settings (similar to application vars without worrying about which env var to use!)

I thought enable switch would be sufficient! 
what about leaving Display and View empty which implicitly won't set any vars ?

Reply:
I think as soon as this is added I will be fine removing the OCIO env vars from doing the same by manually setting it on the Application env vars, I don't mind OP exposing that in a friendlier way. My only worry is that even when the env vars set on the session (so Houdini uses it to the extend it adds support of OCIO natively) that OP would intersect that behavior to do something else coming from the settings, in which case my artists who are familiar with OCIO would shout at me that OP is messing up the Houdini + OCIO native workflow (that happened in Nuke with the OCIO overrides at some point). After I checked the Houdini behavior with these env vars I think we are good in this aspect so you get my approval.

We are setting these on the application DCC settings as opposed to the Project as we want different default view/display per DCC. The $OCIO is ALWAYS set globally for the whole project

Reply:
I came up with a solution where OP extends `OCIO_ACTIVE_DISPLAYS` and `OCIO_ACTIVE_VIEWS` if existed. 
here are some tests.
 
https://github.com/ynput/OpenPype/pull/5858#issuecomment-1807071160